14-reduxy-actions-reducers/README.md

```diff
@@ -0,0 +1,116 @@
+
+In order to do this we will need to separate out our `action-reducers` into two separate files: `actions/index.js` and `reducers/index.js`. Our `actions` will respond to user interactions and communicate to our `reducers` what changes need to occur. The `reducers` will listen for dispatched `actions`, and respond when appropriate, modifying the state as appropriate (the same as before!). This 'broadcasting' and 'listening' is achieved by wrapping our `actions` in a `dispatch()` function.
+
+The goal of this step in particular is to pull out the `actions` and `reducers` into their own specific files and set up the interaction between the two.
+
+One last caveat is our app is very API call heavy, so we need support for asynchronous actions, something vanilla `Redux` does not support out of the box. Enter `redux-thunk`! Redux thunk lets us call `actions` as `actionCreators` like so:
+```js
+const myAction = (argumentFromComponent) => (
+    (dispatch) => (
+        makeApiCall(argumentFromComponent)
+            .then(dispatch(actionToModifyStore))
+    )
+)
+```
+Using syntax similar to above we are able to dispatch an action to update our state upon completion of the API call.
+
+Lets implement this in our app.
+
+As always, if you run into trouble with the [tasks](#tasks) or [exercises](#exercises), you can take a peek at the final [source code](src/).
+
+## Tasks
+
+Create 2 new folders and files `actions/index.js` and `reducers/index.js`.
+
+Now lets separate the markRead and markUnread action-reducers in their respective actions. In our empty actions/index.js import the `setUnread` api call. Next, lets start by bringing over `markRead` and `_setUnread` functions.
+
+The two functions are going to be similar to how they were previously defined, with some key differences:
+
+* Remove emails from both function's arguments
+* Change `markRead` into a redux-thunk style function and pass dispatch into `_setUnread`
+* Remove the state update logic from `_setUnread` and replace with a call to `dispatch()`
+
+Lastly, lets define the traditional redux style action which will be dispatched to update the state of
+the application.
+
+```js
+import {setUnread as setUnreadApi} from '../api';
+
+const SET_UNREAD = 'setUnread';
+const setUnread = (emailId, unread) => ({
+    type: SET_UNREAD,
+    payload: {
+        emailId,
+        unread
+    },
+})
+
+const _setUnread = (dispatch, emailId, unread) => (
+    setUnreadApi(emailId, unread).then(({success}) => {
+        if (success) {
+            return dispatch(actionToBeDefined)
+        }
+
+        throw new Error(
+            `Unable to set email ID# ${emailId} unread state to ${unread}.`
+        );
+    }
+)
+
+export const markRead = (emailId) => (dispatch) => _setUnread(dispatch, emailId, false)
+```
+
+Next lets set up our reducer to handle updating the 'read' state of an email.
+
+Reducers are simply functions which take in the current state, update it if necessary, and
```

Comment:
Don't really need the linebreaks in the paragraph

Reply:
Will remove